src/main/java/com/dabsquared/gitlabjenkins/GitLabPushRequest.java

```diff
@@ -23,6 +27,25 @@ public static GitLabPushRequest create(String payload) {
     public GitLabPushRequest() {
     }
 
+    private GitlabProject sourceProject = null;
+
+    public GitlabProject getSourceProject (GitLab api) throws IOException {
+        if (sourceProject == null) {
+            sourceProject = api.instance().getProject(project_id);
+        }
+        return sourceProject;
+    }
+
+    public GitlabCommitStatus createCommitStatus(GitlabAPI api, String status, String targetUrl) {
+        try {
+            if(getLastCommit()!=null) {
+                return api.createCommitStatus(sourceProject, checkout_sha, status, checkout_sha, "Jenkins", targetUrl, null);
```

Comment:
The same as in `GitlabMergeRequest.java`: Why is `this.checkout_sha` passed as `ref`parameter instead of `this.ref`?